source/firstprogram.rst

```diff
@@ -1,402 +1,621 @@
-Writing your First Program
-==========================
+Writing your First Program:
+===========================
 
-This section is a detailed guide on how to build a `SIR Model <link for SIR URL>`_ in BharatSim from scratch. By the end of this section, you should be able to write and execute a SIR model by yourself.
+This section is a detailed guide for a novice user on how to build a `SIR Model <https://bharatsim.readthedocs.io/en/latest/epidemiology-sir-model.html>`_ in BharatSim from scratch. By the end of this section, you should be able to write and execute a SIR model by yourself.
 
-Any model built using this framework contains different classes which are essentially extensions of different `Nodes <#>`_. To properly build a SIR model from scratch, you will first need to define these classes and the properties associated with them. These classes are grouped under different components of the model. Here, the SIR model has two different components:-
-1. `Agent <#>`_
-2. `Network <#>`_
+Any model built on this framework contains various classes which are essentially extensions of different Nodes. To build a SIR model from scratch, one needs to define the said classes and the properties and relationships associated with them. In the basic SIR there are two main components:

```

Comment:
`Node` is a framework-defined quantity. Keep the link (even if it doesn't work) since we will have to link it to the appropriate page once it's written. For now, you can use a dummy link.

Reply:
@YerikSingh Fix this, please?